sdk/healthinsights/health-insights-cancerprofiling-rest/samples-dev/sample_infer_cancer_profiling.ts

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Infer key cancer attributes such as tumor site, histology, clinical stage TNM categories and pathologic stage TNM
+ *  categories from a patient's unstructured clinical documents.
+ *
+ * @summary detects change points.
+ */
+
+import { AzureKeyCredential } from "@azure/core-auth";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+import CancerProfilingRestClient, {
+    getLongRunningPoller,
+    isUnexpected, OncoPhenotypeData,
+    OncoPhenotypeResultOutput
+} from "../src";
+dotenv.config();
+
+// You will need to set this environment variables or edit the following values
+const endpoint = process.env["HEALTH_INSIGHTS_ENDPOINT"] || "https://eastus.api.cognitive.microsoft.com";
+const apiKey = process.env["HEALTH_INSIGHTS_API_KEY"] || "";
+
+function printResults(cancerProfilingResult: OncoPhenotypeResultOutput): void {
```

Comment:
Think about these samples as the entry point for new customers to your service. The samples should be easy to read and thoroughly documented.

Can you at least add documentation about what the sample is doing, please? If possible to restructure the body of the sample to reduce the nesting, that would be great

Reply:
```typescript
function printResults(cancerProfilingResult, printPrefix: string): void {
  if (cancerProfilingResult === null || cancerProfilingResult === undefined) {
    return;
  }
  for (const key of Object.keys(cancerProfilingResult)) {
    if (cancerProfilingResult[key]) {
      if (typeof cancerProfilingResult[key] !== "object") {
        console.log(`${printPrefix} ${key}: ${cancerProfilingResult[key]}`);
      } else {
        printResults(cancerProfilingResult[key], printPrefix + " " + key);
      }
    }
  }
}
```
@joheredi  how do you think if we print the result like this way ? 
![Uploading image.png…]()

Reply:
Hi @qiaozha .
Thanks for proposing a new way of printing the values .
However, I think that in this sample we would like to demonstrate how to access/iterate over the patients and inferences.
Also to demonstrate how to iterate over the evidence of each inference.
The idea behind this sample is not just dumping the values.

I think that the current sample demonstrates this hierarchy better, even if it's not classic in dumping key, value pairs.